### ISSUE TITLE
React 0.14 handling breaks when itemsRenderer returns a composite element

### ISSUE THREAD

**Issue by taion** [orgsync/react-list#51]
This wrapper causes problems for me: https://github.com/orgsync/react-list/blob/0.6.5/react-list.es6#L10

The issue is that when `itemsRenderer` returns a composite element rather than a DOM element (i.e. something from a custom React component), the special-case React 14 logic in `getEl` does not give the underlying DOM node, and subsequent calls fail.

Personally, I'd be okay with eating the warnings from using `React.findDOMNode` instead of `ReactDOM.findDOMNode` for the time being. Some dynamic require handling might also do the trick, but I only use webpack, so I wouldn't know how to do so in a way that doesn't break people using other bundlers.

Right now I have to work around this with:

```js
renderItems(items, ref) {
  return (
    <div ref={ref}>
      <ListGroup>
        {items}
      </ListGroup>
    </div>
  );
}
```

**Comment by caseywebdev** [comment]
Yea this is kinda weird at the moment because I'd like to support both `0.13.x` and `0.14.x`. I think once `0.15.x` or `1.x` is out I'll drop the `0.13.x` support and then we could safely require `react-dom` and use the appropriate `ReactDOM.findDOMNode`.

As another workaround without using an extra element, you could pass the ref through your `ListGroup` via `listRef={ref}` or something like that, and set it on the top level DOM element inside `ListGroup`.

**Comment by taion** [comment]
IMO just using `React.findDOMNode` would be the ideal solution for React 0.14 pre-release support. The deprecation warning is not that big a deal (to me at least).

You could also do:

```js
let findDOMNode;
if (React.version < '0.14.0') {
  {findDOMNode} = React;
} else {
  {findDOMNode} = require('react-dom');
}
```

I just checked that this pattern doesn't cause issues with webpack, at least.

**Comment by caseywebdev** [comment]
Good suggestion, I slightly modified it to be more conservative and account for browser globals. This is released in `0.6.6`.

**Comment by taion** [comment]
:+1: thanks
